File: Problems/binary_concatenation.py
T = int(input())

def binary_concat(x,y):
    binx = bin(x).replace("0b","")
    biny = bin(y).replace("0b","")
    binxplusY = binx+biny
    binyplusX  = biny+binx
    XplusY = int(binxplusY,2)
    YplusX = int(binyplusX,2)

    # print(XplusY,YplusX)
    return XplusY-YplusX



for i in range(T):
    N = int(input())
    l = list(map(int,input().split())) 
    
    maximum = 0 

    for i in range(len(l)):
        for j in range(len(l)):
            if(i!=j):
                if(maximum <= binary_concat(l[i],l[j])):
                    maximum = binary_concat(l[i],l[j])
    print(maximum)
    
        




# j =list(map(int,input().split()))
# l =[bin(i).replace("0b","") for i in j]

# print(j)
# print(l)

# l = list(map(bin().replace("0b",""),input().split()))
# print(l)

File: Problems/binary_concatenation.cpp
// #include<iostream>
// using namespace std;

// long long binary_concat(long long x,long long y)
// {
//     string binx = bitset
// }

// int main()
// {
//     int T;

// }

#include<iostream>
#include<bits/stdc++.h>
using namespace std;

int binary_concat(int x,int y)
{
    string binx = bitset<64> (x).to_string();
    auto loc1 = binx.find('1');
    if(loc1 != string::npos)
    {
        binx=binx.substr(loc1);
    }

    string biny = bitset<64> (y).to_string();
    auto loc2 = biny.find('1');
    if(loc2 != string::npos)
    {
        biny=biny.substr(loc2);
    }

    string binxplusy = binx+biny;
    string binyplusx = biny+binx;

    int xplusy = stoi(binxplusy,nullptr,2);
    int yplusx = stoi(binyplusx,nullptr,2);
    
    return xplusy-yplusx;
}

int main()
{
    int T,N;
    cin>>T;
    while(T--)
    {
        cin>>N;
        vector<int> V(N);
        int i=0;
        while(N--)
        {
            cin>>V[i];
            i++;
        }
        int maximum = 0;

        for(int i=0;i<V.size();i++)
        {
            for(int j=0;j<V.size();j++)
            {
                if(i != j)
                {
                    if(maximum <= binary_concat(V[i],V[j]))
                    {
                        maximum = binary_concat(V[i],V[j]);
                    }
                }
            }
        }
        cout<<maximum<<endl;
    }
}